app/routes/cron_routes.py

```diff
@@ -1,10 +1,102 @@
+from datetime import datetime
 from flask import Blueprint, request, jsonify
-from app.services import leaderboard_collection_manager
+from app.services import leetcode_review_type_manager, review_manger, fsrs_scheduler, submission_manager
 
 bp = Blueprint('crons', __name__, url_prefix='/tasks')
 
 @bp.route('/review/daily', methods=['GET'])
 def daily_task():
+    '''
+    1. Fetch all collections from users_problemtypes_for_review collection
+    2. For each collection, iterate through each problem types
+    3. For each problem in problem type, update the review_date
+        if the review date has passed and the user has reviewed the problem then the review date should increase
+        if the review date has passed and the user has not reviewed the problem then the review date should decrease
+        there may exist a middle option. The spaced repetition algorithm should be used to determine the next review date
+    4. Update the review date in the document
+    
+    5. If the user has reviewed all problem for that problem type, and the review dates are all
+    above a week, then add a new problem type to the user's review types
+    
+
+
+    {uuid:
+        dP: {problem 1, problem 2, problem 3}
+        DFS: {problem 1, problem 2, problem 3}
+    
+    uuid2
+        
+    }
+
+    Alternatively:
+    
+    Get and create some Metadata
+    1. Fetch the problemtypes collection
+    2. Load all problem data in a hashmap
+        1. Keep a map of problem types to problems
+
+
+    1. Iterate through each users leetcode submissions
+        1. Filter and group the problems by problem type
+        2. Iterate through each problem in the problem type
+            1. Do the logic as before
+
+            
+    Actually, use submission manager to fetch the problems and filter by 'type' field
```

Comment:
Leaving this for right now, just documentation but will remove later + future ideas